doc/languages-frameworks/vscode.section.md

```diff
@@ -0,0 +1,121 @@
+# Visual Studio Code {#visual-studio-code}
+
+_For the free distribution Visual Studio Code (VSC) without MS branding/telemetry see [vscodium](https://nixos.wiki/wiki/VSCodium)_
+
+To install VSC just add it to `systemPackages`:
+
+```nix
+{ environment.systemPackages = with pkgs; [ vscode ]; }
+```
+
+You'll likely want to install some extensions though. For this replace `vscode` with `vscode-with-extensions` and override the `vscodeExtensions` attribute:
+
+```nix
+let
+  extensions = (with pkgs.vscode-extensions; [
+    bbenoist.Nix
+  ]) ++ pkgs.vscode-utils.extensionsFromVscodeMarketplace [{
+    name = "remote-ssh-edit";
+    publisher = "ms-vscode-remote";
+    version = "0.47.2";
+    sha256 = "1hp6gjh4xp2m1xlm1jsdzxw9d8frkiidhph6nvl24d0h8z34w49g";
+  }];
+  vscode-with-extensions = pkgs.vscode-with-extensions.override {
+    vscodeExtensions = extensions;
+  };
+in
+{
+  environment.systemPackages = [
+    vscode-with-extensions
+  ];
+}
+```
+
+You can also install it through [Home Manager](https://nix-community.github.io/home-manager/options.html#opt-programs.vscode.enable):
+
+```nix
+{ programs.vscode.enable = true; }
+```
+
+## Extensions
+
+There are two ways of installing VSC extensions through Nix, and you'll typically use both. The first is to install them as [Nix packages](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&query=vscode-extensions). This is especially important for extensions that require external dependencies, such as `vsliveshare`. The second is to install them directly from the VSC marketplace, in the form of a list of attribute sets.
+
+Lastly, you can also install extensions imperatively (through the VSC UI) just like you would without Nix. This works best if you're not using NixOS, since, on NixOS, it'll fail for extensions such as `vsliveshare`.
+
+The resulting list of extensions can be passed to both `vscodeExtensions` and the `extensions` option from HM. It's recommended to use `vscodeExtensions` though, since the HM option uses an older method of handling extensions, which is not as reliable (see this [issue](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&query=vscode-extensions)).
```

Comment:
You did not link the issue but the search page for vscode extensions.

Reply:
:facepalm: